### DISCUSSION TITLE
Map feature not finding pictures until moved with third party app or backing up

### DISCUSSION THREAD

**Discussion by tibert01** [ente-io/ente#851] [Enhancements]
**Describe the bug**
The map feature, on my device, does not seem to be able to detect pictures on the device. 

Only after moving a picture with an app like Aves (open source gallery) or backing up a picture, will the map find that only picture which was moved or backed up. 

How I moved the picture : move to another folder (pictures/screenshots for example) then back to DCMI/Camera folder.
So the picture did not change location. But it still wasn't detected before. Only after the operation. 

Removing the picture from Ente backup will make it disappear from the map. Even if it still on the device. 

Uninstalling the app and installing it back will make it not detect again the pictures, even the one moved and detected before. 

I have a oneplus device with Oxygen os 13.1 ("color os" skin).
And I was using Google photos for backup. I had only Google photos and the system gallery app.

Moving the picture with the system gallery app doesn't allow ente to detect it.

I am not sure for the steps to reproduce. 

Here is a picture u have with exif location data 
![IMG_20200821_203234](https://github.com/ente-io/photos-app/assets/58490865/f9f2ed41-aa6d-4b8e-af58-638b10a4abe0)
(should be in Barcelona) 

**To Reproduce**
1. Have pictures on the device before installing the app
2. Install app
3. Skip backup
4. Go to map
5. Error message

**Expected behavior**
The app should be able to detect or refresh to see find all the available pictures with 

**Screenshots**
![Screenshot_2023-11-22-10-17-31-15_d55010a4984821e116b93e12b0f0bc31](https://github.com/ente-io/photos-app/assets/58490865/3532ff7e-aa62-4fbc-a49a-c685952e2e82)

**Smartphone (please complete the following information):**
 - Device: OnePlus 8 pro
 - OS: Oxygen Os 13.1 (android 13) patch IN2023_13.1.0.582(EX01) 
 - App Version 0.8.0 F-droid

**Comment by tibert01** [comment]
I checked for updates on the github, and saw the 0.8.1 version.

I installed it (after removing the older version), and it is still happening.

**Comment by tibert01** [comment]
Well, answer from the support team :

> So what's happening is that since extraction of location metadata is expensive (and will slow down the app), we extract this information from photos only at the time of uploads. This means that only uploaded photos can be rendered on the map.
>
> If you backup this photo, you will be able to see this on the map. If not, please let me know!

Other apps having the ability to use such feature without much trouble makes this unexpected.

I think this would either need a clarity enhancement, in the error message or somewhere to inform the user.

Also, why would it work for pictured which have been moved by the Aves app.

**Comment by vishnukvmd** [comment]
It's weird that moving the file via Aves changed the behavior.

Currently we parse the location if it's available from Android's MediaStore: https://github.com/ente-io/photos-app/blob/41a38c753f0667b61528f22ebd8dbf31e526dd7f/lib/models/file/file.dart#L75-L76

If not, we parse it from the EXIF during uploads: https://github.com/ente-io/photos-app/blob/41a38c753f0667b61528f22ebd8dbf31e526dd7f/lib/models/file/file.dart#L174-L177

The latter, while expensive, always works.

From the last comment it seems like moving the file via Aves has somehow gotten MediaStore to update the GPS metadata (that we were then able to parse).

One way forward would be to setup background tasks that parses locations from EXIF for un-uploaded files, but given that most customers primarily use Ente for backups, this might not be very valuable. We'll also need to solve for the UX issue of the Map on different devices rendering different items. The latter is do-able, but the former is non-trivial.

So at the moment I don't have a simple solution to recommend, but if you have any thoughts, do share, would be happy to figure out a way forward!
